aptos-move/framework/supra-framework/sources/pbo_delegation_pool.move

```diff
@@ -1737,46 +1719,45 @@ module supra_framework::pbo_delegation_pool {
-                    already_locked = *table::borrow(&pool.principle_stake, delegator);
-                };
+                // Ensure that all stake to be locked is made `active`.
+                if (active < principle_stake) {
+                    // The amount to lock can be covered by reactivating some previously unlocked stake.
+                    // Only reactivate the required amount to avoid unnecessarily interfering with 
+                    // in-progress withdrawals.
+                    let amount_to_reactivate = principle_stake - active;
+
+                    // Ensure that we do not try to reactivate more than the available `pending_inactive` stake.
+                    // This should be enforced by functions within `authorized_reactivate_stake`, but checking
+                    // again here makes the correctness of this function easier to reason about.
+                    if (amount_to_reactivate > pending_inactive) {
+                        amount_to_reactivate = pending_inactive;
+                    };
 
-                if (max_stake_to_lock > already_locked) {
-                    let max_new_stake_to_lock = max_stake_to_lock - already_locked;
-                    // All `already_locked` stake must be `active`.
-                    let lockable_active_stake = active - already_locked;
-
-                    // Ensure that all stake to be locked is made `active`.
-                    if (lockable_active_stake < max_new_stake_to_lock) {
-                        // The amount to lock can be covered by reactivating some previously unlocked stake.
-                        // Only reactivate the required amount to avoid unnecessarily interfering with 
-                        // in-progress withdrawals.
-                        let amount_to_reactivate = max_new_stake_to_lock - lockable_active_stake;
-
-                        // Ensure that we do not try to reactivate more than the available `pending_inactive` stake.
-                        // This should be enforced by functions within `authorized_reactivate_stake`, but checking
-                        // again here makes the correctness of this function easier to reason about.
-                        if (amount_to_reactivate > pending_inactive) {
-                            amount_to_reactivate = pending_inactive;
+                    if (amount_to_reactivate > MIN_COINS_ON_SHARES_POOL) {
+                        // Reactivate the required amount of `pending_inactive` stake first.
+                        authorized_reactivate_stake(delegator, pool_address, amount_to_reactivate);
+                    };
+
+                    let active_and_pending_inactive = active + pending_inactive;
+                    
+                    if (active_and_pending_inactive < principle_stake) {
+                        // Need to reactivate some of the `inactive` stake.
+                        let amount_to_withdraw = principle_stake - active_and_pending_inactive;
```

Comment:
you may end up withdrawing less than required if `amount_to_reactivate < MIN_COINS_ON_SHARES_POOL`, though it's just a matter of 1 SUPRA and we can ignore if we want to.

Reply:
Thanks. I've updated this.